client/state/data-layer/wpcom-http/actions.js

```diff
@@ -30,15 +40,49 @@ export const http = ( {
 	onFailure,
 	onProgress,
 	...options,
-}, action = null ) => ( {
-	type: WPCOM_HTTP_REQUEST,
-	body,
-	method,
-	path,
-	query: { ...query, apiVersion },
-	formData,
-	onSuccess: onSuccess || action,
-	onFailure: onFailure || action,
-	onProgress: onProgress || action,
-	options,
-} );
+}, action = null ) => {
+	const actionProperties = {
+		onSuccess: onSuccess || action,
+		onFailure: onFailure || action,
+		onProgress: onProgress || action,
+		query: { ...query }
+	};
+
+	// * check parameters *
+
+	// `path` is not optional and it is a string
+	// `method` is not optional and its value can take either 'GET' or 'POST`
+	if (
+		( ! path || typeof path !== 'string' ) ||
+		( ! method || ( method !== 'GET' && method !== 'POST' ) )
+	) {
+		return badRequest();
+	}
+
+	// - One of `apiVersion` of `apiNamespace` should be defined
+	// - `apiVersion` and `apiNamespace` cannot be defined simultaneously
+	if (
+		( typeof apiVersion === 'undefined' && typeof apiNamespace === 'undefined' ) ||
+		( typeof apiVersion !== 'undefined' && typeof apiNamespace !== 'undefined' )
+	) {
+		return badRequest();
+	}
+
+	// - `apiVersion` must have the `v1, v2, ...` shape
+	if ( typeof apiVersion !== 'undefined' && ( /^v\d+/ ).test( apiVersion ) ) {
+		actionProperties.query.apiVersion = apiVersion;
+	}
+
+	// - `apiNamesapce` must have the `wp/v2, wpcom/v2, ...` shape
```

Comment:
@nylen or @retrofox Could we get a bit more clarification on what these namespaces do?

I assume:
- `wp` = WordPress Rest API
- `wpcom` = WordPress.com API

And for anything that has a namespace, any version property is ignored, since that namespace includes a version?

Normally if we are hitting completely different endpoint apis I'd expect 2 client libs, but I'm sure we can clean up the interface so that this is more clear

As @retrofox pointed out earlier it appears that the proxy is only doing a truthy check, which can lead to bugs if you explicitly set a namespace that isn't wp
https://github.com/Automattic/wpcom-proxy-request/blob/master/index.js#L429